src/System.Runtime.Analyzers/Core/SpecifyIFormatProvider.cs

```diff
@@ -156,17 +158,17 @@ public override void Initialize(AnalysisContext analysisContext)
                     }
                     #endregion
 
+
                     #region "IFormatProviderAlternateStringRule & IFormatProviderAlternateRule"
-                    IEnumerable<IMethodSymbol> methodsWithSameNameAsTargetMethod = targetMethod.ContainingType.GetMembers(targetMethod.Name).OfType<IMethodSymbol>();
+
+                    IEnumerable<IMethodSymbol> methodsWithSameNameAsTargetMethod = targetMethod.ContainingType.GetMembers(targetMethod.Name).OfType<IMethodSymbol>().WhereMethodDoesNotContainAttribute(obsoleteTypeAttribute).ToList();
                     if (methodsWithSameNameAsTargetMethod.Count() > 1)
                     {
-                        var correctOverloads = methodsWithSameNameAsTargetMethod.GetMethodOverloadsWithDesiredParameterAtLeadingOrTrailing(targetMethod, iformatProviderType);
+                        var correctOverloads = methodsWithSameNameAsTargetMethod.GetMethodOverloadsWithDesiredParameterAtLeadingOrTrailing(targetMethod, iformatProviderType).ToList();
```

Comment:
In several places in this patch I add `.ToList()`. This is trade memory usage (negligible in these cases) for increased performance, by preventing re-enumeration of possibly complex operations (finding overloads).